tests/README.md

```diff
@@ -299,111 +355,108 @@ Notice we didn't use full rule name on the command line. The prefix `xccdf_org.s
+In the rule mode, the test suite follows the `profiles` metadata key from the
+scenario headers. It will run test scenario for each profile listed in this
+`profile` key. If there is no `profiles` key in the test scenario header the
+test suite will use the virtual `(all)` profile. It is a profile that contains
+all the rules.
 
+Moreover, there is the `--profile` option which can be used to override the
+profile selection from profiles metadata so every test scenario will be executed
+only against the profile selected by this command-line argument and variable
+selections will be done according to this profile.
 
-Scenarios are currently supported only in `bash`. And type of scenario is
-defined by its file name.
+### Debug mode
 
-#### (something).pass.sh
+Use `--debug` option, to investigate a test scenario which is not evaluating to expected result.
+The Test Suite will pause its execution, and you will be able to SSH into the environment to inspect its state manually.
 
-Success scenario - script is expected to prepare machine in such way that the
-rule is expected to pass.
+## Profile-based testing
 
-#### (something).fail.sh
+In this operation mode, you specify the `profile` command and you supply the
+profile ID as a positional argument.  The test suite then runs scans over the
+target domain and remediates it based on particular profile.
 
-Fail scenario - script is expected to break machine so the rule fails. Test
-Suite than, if initial scan fails as expected, tries to remediate machine and
-expects evaluation to pass.
+To test RHEL7 STIG Profile on a VM:
+```
+./test_suite.py profile --libvirt qemu:///session ssg-test-suite-rhel7 --datastream ../build/ssg-rhel7-ds.xml stig
+```
 
-#### (something).sh
+To test Fedora Standard Profile on a Podman container:
+```
+./test_suite.py profile --container ssg_test_suite --datastream ../build/ssg-fedora-ds.xml standard
+```
 
-Support files, these are available to the scenarios during their runtime. Can
-be used as common libraries.
+To test Fedora Standard Profile on a Docker container:
+```
+./test_suite.py profile --docker ssg_test_suite --datastream ../build/ssg-fedora-ds.xml standard
+```
 
-## Combined testing mode
+Note that `profile-id` is matched by the suffix, so it works the same as in `oscap` tool
+(you can use `oscap info --profiles` to see available profiles).
 
-In this mode, you are testing the rules selected by a profile, using the contexts provided by each rule's test scenarios.
-This mode provides an easy way of performing rule-based testing on all rules that are part of a profile.
+### Unselecting problematic rules
 
-Any `# profiles` metadata in the test scenarios will be ignored, the Test Suite will assume that the test scenarios apply for the Profile, since the rule is selected by the profile.
+Sometimes you would like to skip a rule in the profile because they are too slow to test,\
+or you know a rule doesn't have a remediation and you get less value by testing it.
```

Comment:
I would also add one more example case here - some rules might break the test suite, for example the rule `sshd_disable_root_login` which disables root login to the tested VM will break our test suite as it uses root user with all ssh commands (yes we should think about using some user in wheel group in the future so we don't need to unselect such rules).